src/binding_utils.rs

```diff
@@ -349,6 +348,60 @@ macro_rules! impl_enum_field {
     };
 }
 
+macro_rules! _cmd_error_handler {
+    ("handle_bad_timestamp", $cmd_name: path, $rep_type: ident, $rep: ident) => {
+        ::paste::paste! {
+            if let $cmd_name::Rep::BadTimestamp { .. } = $rep {
+                let rep = ::pyo3::Python::with_gil(|py| {
+                    let rep = crate::binding_utils::py_object!(
+                        $rep,
+                        $rep_type,
+                        [<$rep_type BadTimestamp>],
+                        py
+                    );
+                    // `py_object!` uses ? operator, so `with_gil` must return a `Result`
+                    PyResult::Ok(rep)
+                })?;
+                return Err(crate::backend_connection::BackendOutOfBallparkError::new_err(rep));
+            }
+        }
+    };
+}
+
+macro_rules! send_command {
+    ($client: ident, $req: ident, $cmd_name: path, $rep_type: ident, $($kind_type: ty),* $(, $error_handler: literal)? $(,)?) => {
+        // We invoke gil because it will run in a coroutine
+        ::paste::paste! {
+            {
+                let rep = $client
+                    .send($req)
+                    .await
+                    .map_err(|e| ::pyo3::PyErr::from(crate::backend_connection::CommandExc::from(e)))?;
+
+                $(crate::binding_utils::_cmd_error_handler!($error_handler, $cmd_name, $rep_type, rep);)?
+
+                Ok(match rep {
+                    $(
+                        $cmd_name::Rep::$kind_type { .. } => {
+                            ::pyo3::Python::with_gil(|py| {
+                                let rep = crate::binding_utils::py_object!(
+                                    rep,
+                                    $rep_type,
+                                    [<$rep_type $kind_type>],
+                                    py
+                                );
+                                // `py_object!` uses ? operator, so `with_gil` must return a `Result`
+                                PyResult::Ok(rep)
```

Comment:
```suggestion
                                let rep = crate::binding_utils::py_object!(
                                    rep,
                                    $rep_type,
                                    [<$rep_type $kind_type>],
                                    py
                                )?;
                                PyResult::Ok(rep)
```
`py_object` macro return a code block, you should be able to unwrap the result with `?`.

Reply:
It returns a code block but not a result